src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/EmbeddingPreview.jsx

```diff
@@ -40,99 +41,131 @@ import LabelsDesign from '../../../plots-and-tables/components/LabelsDesign';
 const { Panel } = Collapse;
 
 const EmbeddingPreview = () => {
-  const dispatch = useDispatch();
-
   const router = useRouter();
   const { experimentId } = router.query;
   const [selectedSpec, setSelectedSpec] = useState('sample');
   const [plotSpec, setPlotSpec] = useState({});
-  const [config, setConfig] = useState(null);
 
-  const embeddingConfig = useSelector((state) => state.experimentSettings.processing);
-  const embedding = useSelector((state) => state.embeddings);
-  const embeddingMethod = embeddingConfig?.configureEmbedding?.embeddingSettings?.method;
+  const dispatch = useDispatch();
 
   const error = false;
 
   const plots = {
     sample: {
-      title: 'Samples',
-      initialConfig: initialPlotConfigStates.embeddingCategorical,
+      title: 'Colored by Samples',
       specGenerator: generateEmbeddingCategoricalSpec,
       imgSrc: plot1Pic,
+      plotUuid: 'embeddingPreviewBySample',
+      plotType: 'embeddingCategorical',
     },
     cellCluster: {
-      title: 'Default clusters',
-      initialConfig: initialPlotConfigStates.embeddingCategorical,
+      title: 'Colored by CellSets',
       specGenerator: generateEmbeddingCategoricalSpec,
       imgSrc: plot1Pic,
+      plotUuid: 'embeddingPreviewByCellSets',
+      plotType: 'embeddingCategorical',
     },
     mitochondrialFraction: {
       title: 'Mitochondrial fraction reads',
-      initialConfig: initialPlotConfigStates.embeddingContinuous,
       specGenerator: generateEmbeddingContinuousSpec,
       imgSrc: plot2Pic,
+      plotUuid: 'embeddingPreviewMitochondrialReads',
+      plotType: 'embeddingContinuous',
     },
     doubletScore: {
       title: 'Cell doublet score',
-      initialConfig: initialPlotConfigStates.embeddingContinuous,
       specGenerator: generateEmbeddingContinuousSpec,
       imgSrc: plot2Pic,
+      plotUuid: 'embeddingPreviewDoubletScore',
+      plotType: 'embeddingContinuous',
     },
   };
 
+  const config = useSelector((state) => state.componentConfig[plots[selectedSpec].plotUuid]?.config);
+
+  // Prepare data for categorical embedding
+  const embeddingType = 'umap';
```

Comment:
change this to not be hardcoded, but to take the type from the configuration settings

Reply:
Fixed in new commit

Reply:
After refactoring, the call to get `embeddingType` is called in done inside the refactored components.